### DISCUSSION TITLE
Can I use it in Windows?

### DISCUSSION THREAD

**Discussion by hp77-creator** [coder/code-server#2060] [Q&A] [answered]
I read the install guide, Windows wasn't mentioned still I went ahead and tried to install it as I had sh and curl while installing the shell returns Unsupported OS MINGW64_NT-10.0-18363. Is there any alternative other than changing OS or machine?

**Comment by code-asher** [comment]
On idea is to install code-server with `yarn` or `npm` however we don't test on Windows at the moment so if you run into any bugs please report them!

**Reply by geenva** [reply]
That failed with error:
```
internal/modules/cjs/loader.js:968
  throw err;
  ^

Error: Cannot find module 'C:\Users\Marcus\AppData\Roaming\npm\node_modules\code-server\out\node\entry.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
    at Function.Module._load (internal/modules/cjs/loader.js:841:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```
...when installing via Yarn. NPM seemed to fail even during install.

**Comment by sr229** [comment, marked as answer]
You may run this on Windows natively however code-server wasn't exactly build with Windows in mind. Try using WSL1/WSL2 if you want to use Windows for your workflow.

**Reply by hp77-creator** [reply]
Thanks for the reply. I installed ubuntu on wsl2 and when I run the command "code ." in a folder it automatically installs the codeserver on wsl2.

**Reply by 8thHalfHour** [reply]
**PLEASE STOP AND READ THIS BEFORE FOLLOWING THE METHOD DESCRIBED BY @hp77-creator ABOVE.**

This does not install code-server (I.e., a hosted instance of VS Code in a browser over your local network), but rather installs the VSCode Remote Development server (VSCode Server — annoying that the names are so similar) so that a locally installed app on Windows (VS Code) can work inside WSL natively (and vice-versa). The instructions from the VSCode WSL Remote Extension page explain this (See step #3 at [Developing in WSL — Open a remote folder or workspace](https://code.visualstudio.com/docs/remote/wsl#_open-a-remote-folder-or-workspace) where it states: “Type `code .` in the terminal. When doing this for the first time, you should see VS Code fetching components needed to run in WSL. This should only take a short while, and is only needed once.”). The VSCode remote extensions (for Docker, WSL, and SSH) require that VS Code actually be installed locally on the Windows machine.

VS Code-server, as implemented by the code-server app as discussed in this discussion group, is quite different. It is VS Code hosted in a browser that is accessible and usable over the internet or a local network; a self-hosted version of Codespaces. Code-server is, for all intents and purposes, a self-hosted cloud-based IDE, like [Eclipse Theia](https://theia-ide.org/), [Atheos](https://www.atheos.io/), [Icecoder](https://icecoder.net/), or [Codiad](http://codiad.com/) — though it is VS Code. Hope this helps someone else not get confused like I initially did by this post.

I am making this comment here so that people like me don’t come here thinking that the method you described here will actually install the self-hosted, browser implementation of code-server (Codespaces). All they will actually get is the remote extension for WSL development installed. To actually install a hosted version of code-server in the context it is intended by this discussion group, I recommend this thread here: [How I run the code-server in WSL2 (Ubuntu 20.04) with port forwarding and SSL](https://github.com/cdr/code-server/issues/1868#issuecomment-667613991). It worked flawlessly for me, and now I can run VSCode in my browser and work with anything anywhere — including, but not limited to, WSL.

FYI: if you have already installed VS Code on Windows and have stopped by this thread to learn how to install the VS Code Remote Server for WSL development and NOT a cloud-based IDE, then @hp77-creator is correct. However, I recommend following the full instructions on the page I linked above, as provided by Microsoft.

**Reply by hp77-creator** [reply]
Thank you so much for the clarification! I think what limitations you mentioned was what I faced and so I changed my base OS to ubuntu and if someone does stay on windows and wants to use code-server, I guess they should follow what you mentioned. Again sorry for the misinformation but at the time of original post, I did not know of any further details.

**Comment by code-asher** [comment]
@geenva 

You might be missing the dependencies required to build the native Node
modules that code-server depends on.

See https://github.com/Microsoft/vscode/wiki/How-to-Contribute#prerequisites

In my case I already had git, npm, yarn, and Node installed but I also
had to run `npm install --global windows-build-tools --vs2017`. After
that I ran `yarn global add code-server` and was able to launch and use
code-server successfully.
